gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/LocalGcdHelper.java

```diff
@@ -62,17 +64,85 @@
   public static final String DEFAULT_PROJECT_ID = "projectid1";
   public static final int PORT = 8080;
   private static final String GCD = "gcd-v1beta2-rev1-2.1.2b";
+  private static final String GCD_VERSION = "v1beta2";
   private static final String GCD_FILENAME = GCD + ".zip";
   private static final String MD5_CHECKSUM = "d84384cdfa8658e1204f4f8be51300e8";
   private static final URL GCD_URL;
+  private static final String GCLOUD = "gcloud";
+  private static final Path INSTALLED_GCD_PATH;
 
   static {
-    try {
-      GCD_URL = new URL("http://storage.googleapis.com/gcd/tools/" + GCD_FILENAME);
-    } catch (MalformedURLException e) {
-      throw new RuntimeException(e);
+    INSTALLED_GCD_PATH = installedGcdPath();
+    if (INSTALLED_GCD_PATH != null) {
+      GCD_URL = null;
+    } else {
+      try {
+        GCD_URL = new URL("http://storage.googleapis.com/gcd/tools/" + GCD_FILENAME);
+      } catch (MalformedURLException e) {
+        throw new RuntimeException(e);
+      }
     }
   }
+  
+  private static Path installedGcdPath() {
+    Path gcloudPath = executablePath(GCLOUD);
+    gcloudPath = (gcloudPath == null) ? null : gcloudPath.getParent();
+    if (gcloudPath == null) {
+      return null;
+    }
+    Path installedGcdPath = gcloudPath.resolve("platform").resolve("gcd");
+    if (Files.exists(installedGcdPath)) {
+      try {
+        String installedVersion = installedGcdVersion();
+        if (installedVersion != null && installedVersion.startsWith(GCD_VERSION)) {
+          return installedGcdPath;
+        }
+      } catch (IOException | InterruptedException ignore) {
+        // ignore
+      }
+    }
+    return null;
+  }
+  
+  private static String installedGcdVersion() throws IOException, InterruptedException {
+    ProcessBuilder processBuilder = new ProcessBuilder()
+      .redirectErrorStream(true);
+    if (isWindows()) {
+      processBuilder.command("cmd", "/C", "gcloud", "version");
+    } else {
+      processBuilder.command("bash", "gcloud", "version");
+    }
+    Process process = processBuilder.start();
+    process.waitFor();
+    try (BufferedReader reader = 
+            new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+      for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+        if (line.startsWith("gcd-emulator")) {
+          String[] lineComponents = line.split(" ");
+          if (lineComponents.length > 1) {
+            return lineComponents[1];
+          }
+        }
+      }
+      return null;
+    }
+
+  }
+  
+  private static Path executablePath(String cmd) {
```

Comment:
You are right. Gcloud on windows should be simply `gcloud`, not `gcloud.cmd`. I do not have a win machine to try ;(